src/main/java/com/minecraftabnormals/abnormals_core/core/util/DataUtil.java

```diff
@@ -106,8 +110,25 @@ public static boolean matchesKeys(ResourceLocation loc, RegistryKey<?>... keys)
 	 */
 	public static void registerAlternativeDispenseBehavior(Item item, BiPredicate<IBlockSource, ItemStack> condition, IDispenseItemBehavior newBehavior) {
 		IDispenseItemBehavior oldBehavior = DispenserBlock.DISPENSE_BEHAVIOR_REGISTRY.get(item);
-		DispenserBlock.registerDispenseBehavior(item, (source, stack) -> {
-			return condition.test(source, stack) ? newBehavior.dispense(source, stack) : oldBehavior.dispense(source, stack);
-		});
+		DispenserBlock.registerDispenseBehavior(item, (source, stack) -> condition.test(source, stack) ? newBehavior.dispense(source, stack) : oldBehavior.dispense(source, stack));
+	}
+
+	/**
+	 * Adds a new {@link JigsawPiece} to a pre-existing {@link JigsawPattern}.
+	 *
+	 * @param toAdd The {@link ResourceLocation} of the pattern to insert the new piece into.
+	 * @param newPiece The {@link JigsawPiece} to insert into {@code toAdd}.
+	 * @param weight The probability weight of {@code newPiece}.
+	 *
+	 * @author abigailfails
+	 * */
+	public static void addToJigsawPattern(ResourceLocation toAdd, JigsawPiece newPiece, int weight) {
+		JigsawPattern oldPool = WorldGenRegistries.JIGSAW_POOL.getOrDefault(toAdd);
+		if (oldPool != null) {
+			oldPool.rawTemplates.add(Pair.of(newPiece, weight));
+			for (int i=0; i<weight; i++) {
+				oldPool.jigsawPieces.add(newPiece);
```

Comment:
Move `oldpool.jigsawPieces` to a local variable.

Reply:
Done in latest commit.